docs/source/filesystems.mdx

```diff
@@ -10,175 +12,187 @@
+>>> pip install adlfs
 ```
 
-### Loading datasets
+2. Define your credentials
 
-When you are ready to use your dataset again, reload it with [`Dataset.load_from_disk`]:
+```py
+>>> storage_options = {"anon": True}  # for anonymous connection
+# or use your credentials
+>>> storage_options = {"account_name": ACCOUNT_NAME, "account_key": ACCOUNT_KEY)  # gen 2 filesystem
+# or use your credentials with the gen 1 filesystem
+>>> storage_options={"tenant_id": TENANT_ID, "client_id": CLIENT_ID, "client_secret": CLIENT_SECRET}
+```
+
+3. Load your FileSystem instance
 
 ```py
->>> from datasets import load_from_disk
->>> from datasets.filesystems import S3FileSystem
+>>> import adlfs
+>>> fs = adlfs.AzureBlobFileSystem(**storage_options)
+```
 
-# create S3FileSystem without credentials
->>> s3 = S3FileSystem(anon=True)  
+## Load and Save your datasets using your cloud storage FileSystem
 
-# load encoded_dataset to from s3 bucket
->>> dataset = load_from_disk('s3://a-public-datasets/imdb/train',fs=s3)  
+### Load datasets into a cloud storage
 
->>> print(len(dataset))
->>> # 25000
-```
+You can load and cache a dataset into your cloud storage by specifying a remote `cache_dir` in `load_dataset`.
+Don't forget to use the previously defined `storage_options` containing your credentials to write into a private cloud storage.
 
-Load with `botocore.session.Session` and custom AWS profile:
+Load a dataset from the Hugging Face Hub (see [how to load from the Hugging Face Hub](./loading#hugging-face-hub)):
 
 ```py
->>> import botocore
->>> from datasets.filesystems import S3FileSystem
+>>> cache_dir = "s3://my-bucket/datasets-cache"
+>>> builder = load_dataset_builder("imdb", cache_dir=cache_dir, storage_options=storage_options)
+>>> builder.download_and_prepare(file_format="parquet")
+```
 
-# create S3FileSystem instance with aws_access_key_id and aws_secret_access_key
->>> s3_session = botocore.session.Session(profile='my_profile_name')
+Load a dataset using a loading script (see [how to load a local loading script](./loading#local-loading-script)):
 
-# create S3FileSystem instance with s3_session
->>> s3 = S3FileSystem(session=s3_session)
+```py
+>>> cache_dir = "s3://my-bucket/datasets-cache"
+>>> builder = load_dataset_builder("path/to/local/loading_script/loading_script.py", cache_dir=cache_dir, storage_options=storage_options)
+>>> builder.download_and_prepare(file_format="parquet")
+```
 
-# load encoded_dataset to from s3 bucket
->>> dataset = load_from_disk('s3://my-private-datasets/imdb/train',fs=s3)  
+Load your own data files (see [how to load local and remote files](./loading#local-and-remote-files)):
 
->>> print(len(dataset))
->>> # 25000
+```py
+>>> data_files = {"train": ["path/to/train.csv"]}
+>>> cache_dir = "s3://my-bucket/datasets-cache"
+>>> builder = load_dataset_builder("csv", data_files=data_files, cache_dir=cache_dir, storage_options=storage_options)
+>>> builder.download_and_prepare(file_format="parquet")
```

Comment:
From a UX perspective it would be cool if i could do 
```python
# singe file
ds = load_dataset("s3://my-bucket/datasets-cache/train.csv",storage_options=storage_options)
# multi
ds = load_dataset("s3://my-bucket/datasets-cache",data_files={"train":["path/to/train.csv"]},storage_options=storage_options)
```
But not sure if this is possible to implement. Thats for example also how `pandas` is doing it 
`data = pd.read_csv('s3://bucket....csv')`

Reply:
Yup this is definitely doable :)
It sounds super intuitive and practical to use, thanks !